thunder/executors/torchex.py

```diff
@@ -928,7 +928,15 @@ def _add_transform(
 
 # Maps exact inputs to truncation division
 def _div_prim_impl(a: Number | torch.Tensor, b: Number | torch.Tensor) -> torch.Tensor:
-    if dtypes.is_exact_dtype(to_dtype(a.dtype)) and dtypes.is_exact_dtype(to_dtype(a.dtype)):
+    def is_exact_number_or_exact_tensor(x):
+        if isinstance(x, (int, bool)):
+            return True
+        # We use PyTorch's dtype attribute (instead of `dtypes.is_exact_dtype`) so that torch.compile on prims.div works.
```

Comment:
Cool; in the future we might want to refactor this into a helper for other functions, too.

fyi @jjsjann123 because with this check may need to account for number proxies in the future, too